onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -443,20 +443,19 @@ OrtStatus* CreateSessionImpl(_In_ const OrtEnv* env, _In_ const OrtSessionOption
 ORT_API_STATUS_IMPL(OrtApis::CreateSession, _In_ const OrtEnv* env, _In_ const ORTCHAR_T* model_path,
                     _In_ const OrtSessionOptions* options, _Outptr_ OrtSession** out) {
   API_IMPL_BEGIN
-  const auto loader = [model_path](InferenceSession& sess) {
-    return sess.Load(model_path);
-  };
-  return CreateSessionImpl(env, options, loader, out);
+  auto sess = onnxruntime::make_unique<onnxruntime::InferenceSession>(
+      options == nullptr ? nullptr : &options->value, model_path, env->loggingManager);
+  return CreateSessionImpl(env, options, std::move(sess), out);
```

Comment:
>std::move(sess) [](start = 41, length = 15)

This does not match the signature. Currently it is by value.

Reply:
(Correct me if I am wrong) With the current signature, I think the std::move() would have worked fine, except that we would have relinquished ownership. Anyway, I changed the signature to a non-const reference and removed the std::move() based on our offline discussion.